src/KernelAbstractions.jl

```diff
@@ -698,6 +698,18 @@ end
 __size(args::Tuple) = Tuple{args...}
 __size(i::Int) = Tuple{i}
 
+"""
+    argconvert(::Kernel, arg)
+
+Convert arguments to the device side representation.
+"""
+argconvert(k::Kernel{T}, arg) where T =
+    error("Don't know how to convert arguments for Kernel{$T}")
+
+# Enzyme support
+supports_enzyme(::Backend) = false
```

Comment:
is this used anywhere, can it be removed?

Reply:
It's just part of the API and allows the user to check for Enzyme support. `argconvert` is used to lower the array types (e.g., `CuArray`, `CudaDeviceMemory` for the backend). The backend has to support this.

Reply:
Sgtm